src/storage/documentService/documentService.js

```diff
@@ -47,19 +68,35 @@ const validateTtl = ttl => {
 
 const uploadDocument = async (
   document,
+  documentId,
   ttl = DEFAULT_TTL,
   network = config.network
 ) => {
   const verificationResults = await verify(document, network);
-  if (!verificationResults.valid) throw new Error("Document is not valid");
+  if (
+    (!documentId && !verificationResults.valid) ||
```

Comment:
i don't understand why we need this 😅

Reply:
In case of opencerts we publish the document first and then we generate the url. So in that case we need to check if its valid or not. While in case of TT we store the document then we publish the documents. in this case we can only check for the hash validation.

Reply:
fixed